pkg/util/templates/templates.go

```diff
@@ -1,26 +1,16 @@
 package templates
 
 import (
-	"fmt"
-
 	"github.com/openshift/route-monitor-operator/pkg/consts/blackbox"
 	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 )
 
-// TemplateForServiceMonitorName return the generated name from the RouteMonitor.
-// The name is joined by the name and the namespace to create a unique ServiceMonitor for each RouteMonitor
-func TemplateForServiceMonitorName(namespace string, name string) types.NamespacedName {
-	serviceMonitorName := fmt.Sprintf("%s-%s", name, namespace)
-	return types.NamespacedName{Name: serviceMonitorName, Namespace: blackbox.BlackBoxNamespace}
-}
-
 // TemplateForServiceMonitorResource returns a ServiceMonitor
-func TemplateForServiceMonitorResource(url, name string) monitoringv1.ServiceMonitor {
+func TemplateForServiceMonitorResource(url string, namespacedName types.NamespacedName) monitoringv1.ServiceMonitor {
```

Comment:
why not use the type https://github.com/openshift/route-monitor-operator/blob/d57e26c1f813ed740eb6b4a1281392d2fc942b75/api/v1alpha1/common_types.go#L4

it would make the code much cleaner

Reply:
sounds like a good idea!

Reply:
oki, will do in another PR then :)

Reply:
as discussed, since we need it in the `types.NamespaceName` type more often it's not worth the effort.